src/udev/udev-builtin-net_id.c

```diff
@@ -282,28 +342,35 @@ static int dev_pci_slot(struct udev_device *dev, struct netnames *names) {
                 goto out;
         }
 
-        FOREACH_DIRENT_ALL(dent, dir, break) {
-                int i;
-                char *rest, *address, str[PATH_MAX];
-
-                if (dent->d_name[0] == '.')
-                        continue;
-                i = strtol(dent->d_name, &rest, 10);
-                if (rest[0] != '\0')
-                        continue;
-                if (i < 1)
-                        continue;
-
-                xsprintf(str, "%s/%s/address", slots, dent->d_name);
-                if (read_one_line_file(str, &address) >= 0) {
-                        /* match slot address with device by stripping the function */
-                        if (strneq(address, udev_device_get_sysname(names->pcidev), strlen(address)))
-                                hotplug_slot = i;
-                        free(address);
+        hotplug_slot_dev = names->pcidev;
+        while (hotplug_slot_dev) {
+                FOREACH_DIRENT_ALL(dent, dir, break) {
+                        int i;
+                        char *rest, *address, str[PATH_MAX];
+
+                        if (dent->d_name[0] == '.')
+                                continue;
+                        i = strtol(dent->d_name, &rest, 10);
```

Comment:
If possible, please use `safe_atoi()`.

Reply:
I'm hesitant to use safe_atoi() here because it passes 0 as the base, allowing strtol() to interpret the slot filename as number in a different base, which could cause the name to change on some system out there.  Most of the kernel code that calls pci_create_slot() sets the slot name to be a number that's been formatted in decimal, so it shouldn't really be an issue, but there is some code in there that sets the slot name to a string that could possibly be "0x5" or something, which would cause udev to behave differently than it does now.

FWIW, I didn't modify that code--that strtol() is exactly what is in udev right now.  It only shows up as an added line in my patch because I indented it to go inside of a loop.  If you think we need more checks after the strtol() than are there, or if you want to go with safe_atoi() anyway, I will be happy to make that change.